### ISSUE TITLE
Certificates for logstash need to be owned by logstash user

### ISSUE THREAD

**Issue by sriese** [outflanknl/RedELK#12]
Filebeat fails to connect to logstash on a vanilla install on Ubuntu 18.04. Solution is to change ownership of the .crt and .key file to the logstash user per: https://github.com/logstash-plugins/logstash-input-beats/issues/197#issuecomment-457938337

```
chown logstash /etc/logstash/certs/elkserver.key
chown logstash /etc/logstash/certs/elkserver.crt
```

Redir logs: /var/log/filebeat/filebeat:
```
INFO    pipeline/output.go:95   Connecting to backoff(async(tcp://<RedELK>:5044))
ERROR   pipeline/output.go:100  Failed to connect to backoff(async(tcp://<RedELK>:5044)): dial tcp <RedELK>:5044: connect: connection refused
INFO    pipeline/output.go:93   Attempting to reconnect to backoff(async(tcp://<RedELK>:5044)) with 1 reconnect attempt(s)
ERROR   pipeline/output.go:100  Failed to connect to backoff(async(tcp://<RedELK>:5044)): dial tcp <RedELK>:5044: connect: connection refused
INFO    pipeline/output.go:93   Attempting to reconnect to backoff(async(tcp://<RedELK>:5044)) with 2 reconnect attempt(s)
INFO    pipeline/output.go:105  Connection to backoff(async(tcp://<RedELK>:5044)) established
```

logstash logs: /var/log/logstash/logstash-plain.log
```
[ERROR][logstash.inputs.beats    ] Looks like you either have a bad certificate, an invalid key or your private key was not in PKCS8 format.
[WARN ][io.netty.channel.ChannelInitializer] Failed to initialize a channel. Closing: [id: 0xb7d6cfd3, L:/<RedLK>:5044 - R:/<Redir>:46340]
java.lang.IllegalArgumentException: File does not contain valid private key: /etc/logstash/certs/elkserver.key
        at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:267) ~[netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:90) ~[netty-all-4.1.18.Final.jar:4.1.18.Final]
        at org.logstash.netty.SslSimpleBuilder.build(SslSimpleBuilder.java:112) ~[logstash-input-beats-5.1.6.jar:?]
        at org.logstash.beats.Server$BeatsInitializer.initChannel(Server.java:131) ~[logstash-input-beats-5.1.6.jar:?]
        at org.logstash.beats.Server$BeatsInitializer.initChannel(Server.java:101) [logstash-input-beats-5.1.6.jar:?]
        at io.netty.channel.ChannelInitializer.initChannel(ChannelInitializer.java:113) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.ChannelInitializer.handlerAdded(ChannelInitializer.java:105) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:606) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.DefaultChannelPipeline.access$000(DefaultChannelPipeline.java:46) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1412) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1131) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:656) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:510) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:423) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:482) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-all-4.1.18.Final.jar:4.1.18.Final]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_191]
Caused by: java.security.KeyException: could not find key file: /etc/logstash/certs/elkserver.key
        at io.netty.handler.ssl.PemReader.readPrivateKey(PemReader.java:114) ~[netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.handler.ssl.SslContext.toPrivateKey(SslContext.java:1014) ~[netty-all-4.1.18.Final.jar:4.1.18.Final]
        at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:265) ~[netty-all-4.1.18.Final.jar:4.1.18.Final]
        ... 20 more
```

**Comment by MarcOverIP** [comment]
You are absolutely right, and your fix is correct as well. This amongst many other things will be fixed in a major new version that we release in a few weeks time.

**Comment by MarcOverIP** [comment]
Fixed in master branch, and in release 0.9
